specifyweb/specify/migrations/0002_geo.py

```diff
@@ -149,18 +150,19 @@ def revert_default_collection_object_types():
             cog_type_picklist.delete()
 
 def set_discipline_for_taxon_treedefs():
-    for treedef in Taxontreedef.objects.all():
-        if treedef.discipline:
-            continue
-        cot = Collectionobjecttype.objects.filter(taxontreedef=treedef).first()
-        if cot:
-            treedef.discipline = cot.collection.discipline
-            treedef.save()
+    taxon_treedefs_without_discipline = Taxontreedef.objects.filter(discipline__isnull=True)
+    collection_object_types = Collectionobjecttype.objects.filter(taxontreedef__in=taxon_treedefs_without_discipline)
+    
+    treedef_to_discipline = {}
+    for cot in collection_object_types:
+        treedef_to_discipline[cot.taxontreedef_id] = cot.collection.discipline
+    
+    for treedef_id, discipline in treedef_to_discipline.items():
+        Taxontreedef.objects.filter(id=treedef_id).update(discipline=discipline)
```

Comment:
(optional) 
See my comment about Queryset.update over Model.save!

Can you also simplify this? 

Ideally, something like the following: 

```py
    for co_type in collection_object_types: 
        co_type.taxontreedef.discipline = co_type.discipline
```

Although regardless, it would be more performant if you perform a single query and update, rather than an update for each `Taxontreedef`.  You can get a `values_list` or similar (maybe with some annotation and F objects) of the Taxontreedef ids from the `collection_object_types` query!